wireless_modules/middle_module/README.md

```diff
@@ -0,0 +1,62 @@
+
+## Getting Started
+
+1. Press clone on the root folder and use it to clone the entire repository on the command line.
+2. Create a ``config.py`` file in the same directory as the contents of this folder (refer below).
+3. The ``config.py`` file should hold the following variables (ensure the name matches):
+- ``ESSID`` : Should be assigned to a string containing the SSID of the network to connect to
+- ``PASSWORD``: Should be assigned to a string containing password of the network
+- ``MQTT_BROKER``: A string containing either the web server name or IP address which acts like the Mqtt broker
+- ``builtin_LED = PIN(2, PIN.OUT)``
+
+## Contents
+- ``boot.py``: Contains function to connect the micro controller to the internet
+- ``main.py``: Processes MQTT messages and sends sensor data
+- ``MQTT_Client_class.py``: Holds the Client class (used to work with MQTT)
+- ``test.py``: A python file (NOT Micropython) used to run on the local computer to test the ``main.py`` and ``MQTT_Client_class.py`` files. It basically sends on the ``/v3/wireless-module/<-id->/start`` topic, to check whether it initiates the ``main.py`` file to send through the data on the topic ``/v3/wireless-module/<-id->/data`` and then send on the topic ``/v3/wireless-module/<-id->/stop`` to check whether it stops the inflow data. 
+	- Note: There's some latency with MQTT and so after the stop topic is used the ``test.py`` finishes running and so may not print the last few data sent by ``main.py``.
+	- Also when running ``test.py`` if you do not receive any messages, re-run because most likely ``main.py`` was not ready to read the message from the start topic when it was sent by ``test.py`` (This may be due to some error and the need for the ESP32 to restart which would take some time)
+
+
+## Dependencies
+- The ``test.py`` requires the ``paho-mqtt`` library installed for Python
+- The other ``.py`` files are Micropython files so needs a board to run on (eg ESP32)
+- Ensure the ``config.py`` file is created locally (see the 'Getting Started' section above)
+
+
+## Steps to test?
+1. Download all the files (clone the root repository)
+2. Write a ``config.py`` file with the relevant variables (see above)
+3. Read this page (especially ensuring you understand what ``test.py`` does and what dependencies exist)
+4. Run the ``boot.py`` file to get your ESP32 connected to the WIFI
+5. Run the ``main.py`` file and after starting it run the ``test.py`` file on your local computer and watch the console.
+6. Sample messages from the ``main.py`` file will be sent over to ``test.py`` which would output it to the console. The ``test.py`` file allows ``main.py`` to send for 10-14 seconds (default).
+7. Note that this process may take around 20 seconds (10 seconds for actual data sending and allowing for 10 seconds initially in case the ESP32 needs to reboot to reconnect to the MQTT broker)
```

Comment:
This largely duplicates the Contents section

Reply:
I tried to remove from these steps which I thought was overstating stuff but didn't really find much that's explicitly duplicated from the Contents section.

Reply:
Looking back, I must have misread the readme. Ignore this!